simulator-starter/src/main/java/com/consol/citrus/simulator/correlation/CorrelationHandlerBuilder.java

```diff
@@ -62,17 +72,21 @@ public CorrelationHandlerBuilder onPayload(String expression, String value) {
     }
 
     public CorrelationHandlerBuilder withHandler(CorrelationHandler handler) {
-        action.setCorrelationHandler(handler);
+		correlationHandler = handler;
         stopCorrelationAction.setCorrelationHandler(handler);
         return this;
     }
 
+	public CorrelationHandler getCorrelationHandler() {
+		return correlationHandler;
+	}
+
     public TestAction stop() {
         return stopCorrelationAction;
     }
 
     private NamespaceContextBuilder lookupNamespaceContextBuilder() {
-        String[] beanNames = BeanFactoryUtils.beanNamesForTypeIncludingAncestors(applicationContext, NamespaceContextBuilder.class);
+    	String[] beanNames = BeanFactoryUtils.beanNamesForTypeIncludingAncestors(applicationContext, NamespaceContextBuilder.class);
```

Comment:
Note that this is the only reason why this builder still has a dependency to applicationContext rather than referenceResolver.

Reply:
I'll take a look at this, if you want me to.

Reply:
Just go ahead.

Reply:
ok, so. that is not really a "problem", right? maybe not that nice. @christophd do you have a static `ApplicationContextHolder` anywhere (which implements `ApplicationContextAware`)? otherwise I'd just keep it this way. however, take a look at the log sequences below - they don't make any sense 😅 should be:

```java
                log.warn("Expected to find 1 beans of type {} but found {} instead: {}",
                    NamespaceContextBuilder.class.getCanonicalName(),
                    beanNames.length,
                    beanNames
                );
```